snuba/query/__init__.py

```diff
@@ -412,6 +416,64 @@ def get_columns_referenced_in_select(self) -> Set[Column]:
             [selected.expression for selected in self.__selected_columns], Column
         )
 
+    def find_and_replace_column_in_condition(
```

Comment:
This logic shouldn't be in the Query class. It should be in the processor, and it might actually be a duplicate of logic we have elsewhere. I'd have to look to see.